File: Java/com.faculdadevincit.designpatterns.adapterpattern/src/XmlToJsonAdapter.java
import org.json.XML;
import org.json.JSONObject;

public class XmlToJsonAdapter implements IJsonMessage{
    private final XmlMessage xmlMessage;
    
    public XmlToJsonAdapter(XmlMessage xmlMessage) {
        this.xmlMessage = xmlMessage;
    }

    @Override
    public String getJson() {
        String xml = xmlMessage.getXml();
        JSONObject jsonObject = XML.toJSONObject(xml);

        return jsonObject.toString(4);
    }

}

File: Python/adapter_pattern/xml_to_json_adapter.py
import json
import xmltodict

from json_message import JsonMessage

class XmlToJsonAdapter(JsonMessage):
    def __init__(self, xml_message):
        self.xml_message = xml_message
        
    def get_json(self):
        xml = self.xml_message.get_xml()
        json_data = json.dumps(xmltodict.parse(xml), indent=4)
        
        return json_data